File: java/1287. Element Appearing More Than 25% In Sorted Array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        int count = 0;
        int current = -1;
        for (int i : arr) {
            if (current == i) {
                count++;
            } else {
                count = 1;
                current = i;
            }
            if (100 * count > 25*arr.length) {
                return current;
            }
        }
        return -1;
    }
}

File: go/1287. Element Appearing More Than 25% In Sorted Array.go
func findSpecialInteger(arr []int) int {
    count := 0
    current := -1
    for _, v := range arr {
        if v == current {
            count++
        } else {
            current = v
            count = 1
        }
        if 100*count > 25*len(arr) {
            return current
        }
    }
    return -1
}
